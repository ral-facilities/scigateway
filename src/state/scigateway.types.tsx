import { Theme } from '@mui/material';
import { Step } from 'react-joyride';
import { LogoState } from './state.types';

export const AuthSuccessType = 'scigateway:auth_success';
export const AuthFailureType = 'scigateway:auth_failure';
export const AutoLoginSuccessType = 'scigateway:autologin_success';
export const LoadingAuthType = 'scigateway:loading_auth';
export const LoadedAuthType = 'scigateway:loaded_auth';
export const LoadAuthProviderType = 'scigateway:loading_auth_provider';
export const LoadScheduledMaintenanceStateType =
  'scigateway:load_scheduled_maintenance_state';
export const LoadMaintenanceStateType = 'scigateway:load_maintenance_state';
export const InvalidateTokenType = 'scigateway:api:invalidate_token';
export const ConfigureStringsType = 'scigateway:configure_strings';
export const LoginType = 'scigateway:login';
export const NotificationType = 'scigateway:api:notification';
export const RegisterRouteType = 'scigateway:api:register_route';
export const RequestPluginRerenderType = 'scigateway:api:plugin_rerender';
export const SendThemeOptionsType = 'scigateway:api:send_themeoptions';
export const LoadDarkModePreferenceType =
  'scigateway:load_dark_mode_preference';
export const LoadHighContrastModePreferenceType =
  'scigateway:load_high_contrast_mode_preference';
export const SignOutType = 'scigateway:signout';
export const ResetAuthStateType = 'scigateway:reset_auth_state';
export const BroadcastSignOutType = 'scigateway:api:signout';
export const ToggleDrawerType = 'scigateway:toggledrawer';
export const DismissNotificationType = 'scigateway:dismissnotification';
export const ConfigureFeatureSwitchesType = 'scigateway:feature_switches';
export const AddNotificationsType = 'scigateway:add_notification';
export const SiteLoadingType = 'scigateway:site_loading';
export const ConfigureAnalyticsType = 'scigateway:configure_analytics';
export const InitialiseAnalyticsType = 'scigateway:initialise_analytics';
export const ToggleHelpType = 'scigateway:toggle_help';
export const AddHelpTourStepsType = 'scigateway:add_help_tour_steps';
export const RegisterHomepageUrlType = 'scigateway:register_homepage_url';
export const CustomLogoType = 'scigateway:custom_logo';
export const CustomNavigationDrawerLogoType =
  'scigateway:custom_navigation_drawer_logo';
export const CustomAdminPageDefaultTabType =
  'scigateway:custom_admin_default_tab';
export const RegisterContactUsAccessibilityFormUrlType =
  'scigateway:contact_us_accessibility_form_url';
export const CustomPrimaryColourType = 'scigateway:custom_primary_colour';

export const scigatewayRoutes = {
  home: '/',
  contact: '/contact',
  help: '/help',
  admin: '/admin',
  login: '/login',
  logout: '/logout',
  cookies: '/cookies',
  accessibility: '/accessibility',
};
export const adminRoutes = {
  maintenance: '/admin/maintenance',
  download: '/admin/download',
};

export interface NotificationPayload {
  message: string;
  severity: string;
}

export interface AddNotificationsPayload {
  notifications: NotificationPayload[];
}

export type AppStrings = Record<string, string>;

export interface ConfigureStringsPayload {
  res: ApplicationStrings;
}

export type ApplicationStrings = Record<string, AppStrings>;

export interface FeatureSwitchesPayload {
  switches: FeatureSwitches;
}

export interface FeatureSwitches {
  showHelpPageButton: boolean;
  singlePluginLogo: boolean;
}

export interface HomepageUrlPayload {
  homepageUrl: string;
}

export interface CustomLogoPayload {
  logo: string;
}

export interface CustomNavigationDrawerLogoPayload {
  navigationDrawerLogo: LogoState;
}

export interface CustomAdminPageDefaultTabPayload {
  adminPageDefaultTab: 'maintenance' | 'download';
}

export interface CustomPrimaryColourPayload {
  primaryColour: string;
}

export interface RegisterRoutePayload {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  unauthorised?: boolean;
  hideFromMenu?: boolean;
  admin?: boolean;
  order: number;
  helpText?: string;
  logoLightMode?: string;
  logoDarkMode?: string;
  logoAltText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface PluginConfig {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  unauthorised?: boolean;
  hideFromMenu?: boolean;
  admin?: boolean;
  order: number;
  helpText?: string;
  logoLightMode?: string;
  logoDarkMode?: string;
  logoAltText?: string;
  helpSteps?: { target: string; content: string }[];
}

export type GroupedPlugins = Record<string, PluginConfig[]>;

export interface LoginPayload {
  username: string;
  password: string;
}

export interface AccessPayload {
  authorisation: boolean;
  accessToken: string;
  error: null;
}

export interface AuthorisedPayload {
  token: string;
}

export interface SendThemeOptionsPayload {
  theme: Theme;
  broadcast: boolean;
}

export interface LoadDarkModePreferencePayload {
  darkMode: boolean;
}

export interface LoadHighContrastModePreferencePayload {
  highContrastMode: boolean;
}

export interface DismissNotificationPayload {
  index: number;
}

export interface AuthProviderPayload {
  authProvider: string | null;
  authUrl?: string;
  autoLogin?: boolean;
}

export interface ScheduledMaintenanceStatePayLoad {
  scheduledMaintenance: ScheduledMaintenanceState;
}

export interface ScheduledMaintenanceState {
  show: boolean;
  message: string;
  severity?: 'success' | 'warning' | 'error' | 'information';
}

export interface MaintenanceStatePayLoad {
  maintenance: MaintenanceState;
}

export interface MaintenanceState {
  show: boolean;
  message: string;
}

export interface SiteLoadingPayload {
  loading: boolean;
}

export interface ConfigureAnalyticsPayload {
  id: string;
}

export interface AddHelpTourStepsPayload {
  steps: Step[];
}

export interface ContactUsAccessibilityFormUrlPayload {
  contactUsAccessibilityFormUrl: string;
}
