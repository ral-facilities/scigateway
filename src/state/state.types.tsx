import { ThunkAction } from 'redux-thunk';
import { AnyAction } from 'redux';
import { Step } from 'react-joyride';
import {
  ApplicationStrings,
  PluginConfig,
  FeatureSwitches,
  ScheduledMaintenanceState,
  MaintenanceState,
} from './scigateway.types';
import { RouterState } from 'connected-react-router';

export interface Plugin {
  name: string;
  src: string;
  enable: boolean;
  unauthorised?: boolean;
  location: 'main' | 'left' | 'right';
}

export interface ScigatewayNotification {
  message: string;
  severity: string;
}

export interface ScigatewayState {
  notifications: ScigatewayNotification[];
  plugins: PluginConfig[];
  drawerOpen: boolean;
  siteLoading: boolean;
  showHelp: boolean;
  helpSteps: Step[];
  authorisation: AuthState;
  res?: ApplicationStrings;
  features: FeatureSwitches;
  analytics?: AnalyticsState;
  darkMode: boolean;
  highContrastMode: boolean;
  homepageUrl?: string;
  logo?: string;
  scheduledMaintenance: ScheduledMaintenanceState;
  maintenance: MaintenanceState;
  navigationDrawerLogo?: LogoState;
  adminPageDefaultTab?: 'maintenance' | 'download';
  contactUsAccessibilityFormUrl?: string;
  primaryColour?: string;
}

export interface StateType {
  scigateway: ScigatewayState;
  router: RouterState;
}

export interface ActionType<T> {
  type: string;
  payload: T;
}

export type ThunkResult<R> = ThunkAction<R, StateType, null, AnyAction>;

export interface User {
  username: string;
  isAdmin: boolean;
  avatarUrl: string;
}

export interface ICATAuthenticator {
  mnemonic: string;
  keys: { name: string; hide?: boolean }[];
  friendly?: string;
  admin?: boolean;
}

export interface AuthProvider {
  isLoggedIn: () => boolean;
  isAdmin: () => boolean;
  logOut: () => void;
  logIn: (username: string, password: string) => Promise<void>;
  verifyLogIn: () => Promise<void>;
  refresh: () => Promise<void>;
  fetchScheduledMaintenanceState?: () => Promise<ScheduledMaintenanceState>;
  setScheduledMaintenanceState?: (
    scheduledMaintenanceState: ScheduledMaintenanceState
  ) => Promise<string | void>;
  fetchMaintenanceState?: () => Promise<MaintenanceState>;
  setMaintenanceState?: (
    maintenanceState: MaintenanceState
  ) => Promise<string | void>;
  redirectUrl: string | null;
  authUrl: string | undefined;
  user: User | null;
  mnemonic?: string;
  autoLogin?: () => Promise<void>;
}

export interface AuthState {
  failedToLogin: boolean;
  signedOutDueToTokenInvalidation: boolean;
  loading: boolean;
  provider: AuthProvider;
}

export interface AnalyticsState {
  id: string;
  initialised: boolean;
}

export interface LogoState {
  light: string;
  dark: string;
  altTxt: string;
}
