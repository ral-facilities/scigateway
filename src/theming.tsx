import {
  createTheme,
  ThemeProvider,
  StyledEngineProvider,
  Theme,
} from '@mui/material/styles';
import 'react-redux-toastr/lib/css/react-redux-toastr.min.css';
import React from 'react';
import { StateType } from './state/state.types';
import { connect, useSelector } from 'react-redux';
import { checkboxClasses } from '@mui/material/Checkbox';
import {
  formHelperTextClasses,
  formLabelClasses,
  inputClasses,
  outlinedInputClasses,
  tabClasses,
} from '@mui/material';

/* UKRI colours */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const UKRI_COLOURS = {
  blue: '#2E2D62',
  bright: {
    orange: '#FF6900', // pure orange
    yellow: '#FBBB10', // yellow
    green: '#67C04D', // light green
    blue: '#1E5DF8', // blue
    purple: '#BE2BBB', // bright purple
    red: '#E94D36', // light red
  },
  deep: {
    orange: '#C13D33', // pure orange
    yellow: '#F08900', // vivid yellow
    green: '#3E863E', // green
    blue: '#003088', // blue
    purple: '#8A1A9B', // bright purple
    red: '#A91B2E', // red
  },
};

/* Colours that may be used across light/dark modes e.g. the main app bar */
const STATIC_COLOURS = {
  darkBlue: '#003088',
  orange: '#C34F00',
};

/* Main colours used for dark/light modes respectively */
interface ThemeColours {
  /* The type of mode this is */
  type: 'default' | 'contrast';

  /* Primary/secondary colours used for MUI */
  primary: string;
  secondary: string;

  /* Secondary text colour */
  textSecondary: string;

  /* Background colours for the page and papers */
  background: string;
  paper: string;

  /* Standard colours used in plugins (change to lighter/darker shades
    between light and dark modes) - these are meant to give good contrast
    for text on the chosen paper background colour */
  blue: string;
  orange: string;
  red: string;
  grey: string;

  /* Lighter colours */
  lightBlue: string; //Used for lighter coloured buttons
  lightOrange: string; //Used for notifcation icon

  /* Darker colours */
  darkGreen: string; //Used for cookie consent message
  darkOrange: string; //Used for help tour

  /* Contrast colours that need to change significantly between dark
     and light modes */
  contrastGrey: string; //Used on homepage

  /* Informational/Warning colours */
  information: string; //Used in open data label
  warning: string; //Used for selection alert banner

  /* Colours for <a> style links */
  link: {
    default: string;
    visited: string;
    active: string;
  };

  footerLink: {
    default: string;
    active: string;
  };

  chip: string; //Used for chip colours in cards and filters

  /* Colours used on the homepage */
  homePage: {
    heading: string;
    blueDescription: string;
    blueButton: string;
  };

  /* Colour for Tabs in search */
  tabsGrey: string;
}

declare module '@mui/material/styles' {
  interface Theme {
    colours: ThemeColours;
    drawerWidth: string;
    footerPaddingTop: string;
    footerPaddingBottom: string;
    footerHeight: string;
    mainAppBarHeight: string;
  }
  // allow configuration using `createTheme`
  interface ThemeOptions {
    colours: ThemeColours;
    drawerWidth: string;
    footerPaddingTop: string;
    footerPaddingBottom: string;
    footerHeight: string;
    mainAppBarHeight: string;
  }
}

const DARK_MODE_COLOURS: ThemeColours = {
  type: 'default',
  primary: UKRI_COLOURS.deep.blue,
  secondary: '#80ACFF',
  textSecondary: 'rgba(255, 255, 255, 0.7)',
  background: '#1B1B1B',
  paper: '#3A3A3A',
  blue: '#86B4FF',
  orange: '#F26300',
  red: '#FF7F73',
  grey: '#A4A4A4',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#E6E6E6',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#86B4FF',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#595959',
  homePage: {
    heading: '#FFFFFF',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#3A3A3A',
};

const DARK_MODE_HIGH_CONTRAST_COLOURS: ThemeColours = {
  type: 'contrast',
  primary: '#86B4FF',
  secondary: '#80ACFF',
  textSecondary: 'rgba(255, 255, 255, 0.7)',
  background: '#000000',
  paper: '#1A1A1A',
  blue: '#B4CCFA',
  orange: '#FFC14D',
  red: '#FFA198',
  grey: '#A4A4A4',
  lightBlue: UKRI_COLOURS.deep.blue,
  lightOrange: '#FFC14D',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#FFFFFF',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFC14D',
  link: {
    default: '#B4CCFA',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#000000',
    active: '#851D0F',
  },
  chip: '#3A3A3A',
  homePage: {
    heading: '#FFFFFF',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#1A1A1A',
};

const LIGHT_MODE_COLOURS: ThemeColours = {
  type: 'default',
  primary: UKRI_COLOURS.deep.blue,
  secondary: UKRI_COLOURS.deep.blue,
  textSecondary: 'rgba(0, 0, 0, 0.54)',
  background: '#FAFAFA',
  paper: '#FFFFFF',
  blue: UKRI_COLOURS.deep.blue,
  orange: '#C34F00',
  red: '#AC1600',
  grey: '#727272',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#717171',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#1E5DF8',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#E0E0E0',
  homePage: {
    heading: '#333333',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#EEEEEE',
};

const LIGHT_MODE_HIGH_CONTRAST_COLOURS: ThemeColours = {
  type: 'contrast',
  primary: UKRI_COLOURS.deep.blue,
  secondary: UKRI_COLOURS.deep.blue,
  textSecondary: '#000000',
  background: '#FAFAFA',
  paper: '#FFFFFF',
  blue: '#002466',
  orange: '#C34F00',
  red: '#801100',
  grey: '#727272',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#000000',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#052d94',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#E0E0E0',
  homePage: {
    heading: '#000000',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#EEEEEE',
};

export const buildTheme = (
  darkModePreference: boolean,
  highContrastModePreference?: boolean,
  primaryColour?: string
): Theme => {
  const colours = darkModePreference
    ? highContrastModePreference
      ? DARK_MODE_HIGH_CONTRAST_COLOURS
      : DARK_MODE_COLOURS
    : highContrastModePreference
      ? LIGHT_MODE_HIGH_CONTRAST_COLOURS
      : LIGHT_MODE_COLOURS;

  if (primaryColour && !(darkModePreference && highContrastModePreference)) {
    colours.primary = primaryColour;
  }

  const componentOverrides = {
    MuiPaper: {
      styleOverrides: { root: { backgroundImage: 'unset' } },
    },
    MuiLink: {
      defaultProps: {
        underline: 'hover' as 'always' | 'none' | 'hover' | undefined,
      },
      styleOverrides: {
        root: {
          color: colours.blue,
        },
      },
    },
    MuiTabs: {
      styleOverrides: {
        indicator: {
          color: STATIC_COLOURS.darkBlue,
          textDecoration: 'underline',
        },
      },
    },
    MuiTab: {
      styleOverrides: {
        textColorSecondary: {
          [`&.${tabClasses.selected}`]: {
            color: darkModePreference ? '#FFFFFF' : colours.blue,
          },
        },
      },
    },
    MuiCheckbox: {
      styleOverrides: {
        colorPrimary: {
          [`&.${checkboxClasses.checked}`]: {
            color: colours.blue,
          },
          [`&.${checkboxClasses.indeterminate}`]: {
            color: colours.grey,
          },
        },
      },
    },
    MuiFormLabel: {
      styleOverrides: {
        root: {
          [`&.${formLabelClasses.error}`]: {
            color: colours.red,
          },
          [`&.${formLabelClasses.focused}`]: {
            color: colours.blue,
          },
        },
        asterisk: {
          [`&.${formLabelClasses.error}`]: {
            color: colours.red,
          },
        },
      },
    },
    MuiBadge: {
      styleOverrides: {
        colorPrimary: {
          backgroundColor: STATIC_COLOURS.orange,
          color: 'white',
        },
      },
    },
    MuiInput: {
      styleOverrides: {
        underline: {
          [`&.${inputClasses.error}`]: {
            borderBottomColor: colours.red,
          },
          [`&.${inputClasses.error}:after`]: {
            borderBottomColor: colours.blue,
          },
        },
      },
    },
    MuiOutlinedInput: {
      styleOverrides: {
        root: {
          [`&.${outlinedInputClasses.error} .${outlinedInputClasses.notchedOutline}`]:
            {
              borderColor: colours.red,
            },
        },
        //Override opacity of placeholder text
        input: {
          '&::placeholder': {
            colour: colours.textSecondary,
            opacity:
              !darkModePreference && highContrastModePreference ? 1.0 : 0.7,
          },
        },
      },
    },
    MuiFormHelperText: {
      styleOverrides: {
        root: {
          [`&.${formHelperTextClasses.error}`]: {
            color: colours.red,
          },
        },
      },
    },
    MuiChip: {
      styleOverrides: {
        root: {
          backgroundColor: colours.chip,
        },
      },
    },
    MuiPickersToolbar: {
      styleOverrides: {
        toolbar: {
          backgroundColor: colours.primary,
        },
      },
    },
    MuiPickersCalendarHeader: {
      styleOverrides: {
        dayLabel: {
          color: colours.grey,
        },
      },
    },
    MuiPickersDay: {
      styleOverrides: {
        current: {
          color: colours.blue,
        },
        dayDisabled: {
          color: colours.grey,
        },
      },
    },
    MuiPickersYear: {
      styleOverrides: {
        root: {
          '&:active': {
            color: colours.blue,
          },
          '&:focus': {
            color: colours.blue,
          },
        },
        yearSelected: {
          color: colours.blue,
        },
        yearDisabled: {
          color: colours.grey,
        },
      },
    },
    MuiPickersMonth: {
      styleOverrides: {
        root: {
          '&:active': {
            color: colours.blue,
          },
          '&:focus': {
            color: colours.blue,
          },
        },
        monthSelected: {
          color: colours.blue,
        },
        monthDisabled: {
          color: colours.grey,
        },
      },
    },
  };

  return createTheme({
    palette: {
      // Light/dark mode
      mode: darkModePreference ? 'dark' : 'light',
      primary: {
        main: colours.primary,
      },
      secondary: {
        main: colours.secondary,
      },
      text: {
        secondary: colours.textSecondary,
      },
      background: {
        default: colours.background,
        paper: colours.paper,
      },
    },
    typography: {
      button: {
        textTransform: 'none',
      },
    },
    components: componentOverrides,
    colours: colours,
    drawerWidth: '220px',
    footerPaddingTop: '8px',
    footerPaddingBottom: '8px',
    footerHeight: '20px',
    mainAppBarHeight: '64px',
  });
};

function mapThemeProviderStateToProps(state: StateType): {
  prefersDarkMode: boolean;
} {
  return {
    prefersDarkMode: state.scigateway.darkMode,
  };
}

const SciGatewayThemeProvider = (props: {
  children: React.ReactNode;
}): React.ReactElement<{
  children: React.ReactNode;
}> => {
  const darkModePreference = useSelector(
    (state: StateType) => state.scigateway.darkMode
  );
  const highContrastModePreference = useSelector(
    (state: StateType) => state.scigateway.highContrastMode
  );
  const primaryColour = useSelector(
    (state: StateType) => state.scigateway.primaryColour
  );
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider
        theme={buildTheme(
          darkModePreference,
          highContrastModePreference,
          primaryColour
        )}
      >
        {props.children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

export const ConnectedThemeProvider = connect(mapThemeProviderStateToProps)(
  SciGatewayThemeProvider
);
