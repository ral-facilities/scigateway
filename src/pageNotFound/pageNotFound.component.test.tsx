import React from 'react';
import PageNotFoundComponent from './pageNotFound.component';
import { thunk } from 'redux-thunk';
import { authState, initialState } from '../state/reducers/scigateway.reducer';
import { createMemoryHistory, History } from 'history';
import configureStore from 'redux-mock-store';
import { StateType } from '../state/state.types';
import { Provider } from 'react-redux';
import { StyledEngineProvider, ThemeProvider } from '@mui/material';
import { buildTheme } from '../theming';
import { Router } from 'react-router';
import { render, screen } from '@testing-library/react';

describe('Page Not found component', () => {
  let state: StateType;
  let history: History;

  function Wrapper({
    children,
  }: {
    children: React.ReactElement;
  }): JSX.Element {
    return (
      <Provider store={configureStore([thunk])(state)}>
        <Router history={history}>
          <StyledEngineProvider injectFirst>
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </StyledEngineProvider>
        </Router>
      </Provider>
    );
  }

  beforeEach(() => {
    state = {
      scigateway: { ...initialState, authorisation: { ...authState } },
    };

    history = createMemoryHistory();
  });

  const theme = buildTheme(false);

  it('renders pageNotFound page correctly', () => {
    const { asFragment } = render(<PageNotFoundComponent />, {
      wrapper: Wrapper,
    });

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByRole('link', { name: 'homepage' })).toHaveAttribute(
      'href',
      '/'
    );
    expect(
      screen.getByRole('link', { name: 'contact support' })
    ).toHaveAttribute('href', '/footer.links.contact');
  });
});
