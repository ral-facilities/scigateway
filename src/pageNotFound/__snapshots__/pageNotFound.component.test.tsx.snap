// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Not found component renders pageNotFound page correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-9thxj7"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ze4q0y-MuiSvgIcon-root"
        data-testid="BugReportIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1dathbx-MuiTypography-root"
      >
        404
      </p>
    </div>
    <div
      class="MuiBox-root css-3i5n1c"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-18ft2kn-MuiTypography-root"
      >
        page-not-found.title
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-n19zd7-MuiTypography-root"
      >
        We're sorry, the page you requested was not found on the server. If you entered the URL manually please check your spelling and try again. Otherwise, return to the 
        <a
          class="css-1jolfkj"
          data-test-id="page-not-found-homepage-link"
          href="/"
        >
          homepage
        </a>
         or 
        <a
          class="css-1jolfkj"
          data-test-id="page-not-found-contact-support-link"
          href="/footer.links.contact"
        >
          contact support
        </a>
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;
